crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,538 @@
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Result<FixedPoint> {
+        let price = self.calculate_spot_price_after_long(base_amount, Some(bond_amount))?;
+        let price_derivative = self.price_after_long_derivative(base_amount, bond_amount)?;
+        // The actual equation we want to represent is:
+        // r' = -p' / (t \cdot p^2)
+        // We can do a trick to return a positive-only version and
+        // indicate that it should be negative in the fn name.
+        // We use price * price instead of price.pow(fixed!(2e18)) to avoid error introduced by pow.
+        Ok(price_derivative / (self.annualized_position_duration() * price * price))
+    }
+
+    /// The derivative of the price after a long.
+    ///
+    /// The price after a long that moves shares by $\Delta z$ and bonds by $\Delta y$
+    /// is equal to
+    ///
+    /// $$
+    /// p(\Delta z) = (\frac{\mu \cdot (z_{0} + \Delta z - (\zeta_{0} + \Delta \zeta))}{y - \Delta y})^{t_{s}}
+    /// $$
+    ///
+    /// where $t_{s}$ is the time stretch constant and $z_{e,0}$ is the initial
+    /// effective share reserves, and $\zeta$ is the zeta adjustment.
+    /// The zeta adjustment is constant when opening a long, i.e.
+    /// $\Delta \zeta = 0$, so we drop the subscript. Equivalently, for some
+    /// amount of `delta_base`$= x$ provided to open a long, we can write:
+    ///
+    /// $$
+    /// p(x) = (\frac{\mu \cdot (z_{e,0} + \frac{x}{c} - g(x) - \zeta)}{y_0 - y(x)})^{t_{s}}
+    /// $$
+    ///
+    /// where $g(x)$ is the [open_long_governance_fee](long::fees::open_long_governance_fee),
+    /// $y(x)$ is the [long_amount](long::open::calculate_open_long),
+    ///
+    ///
+    /// To compute the derivative, we first define some auxiliary variables:
+    ///
+    /// $$
+    /// a(x) = \mu (z_{0} + \frac{x}{c} - g(x) - \zeta) \\
+    /// b(x) = y_0 - y(x) \\
+    /// v(x) = \frac{a(x)}{b(x)}
+    /// $$
+    ///
+    /// and thus $p(x) = v(x)^t_{s}$. Given these, we can write out intermediate derivatives:
+    ///
+    /// $$
+    /// a'(x) = \frac{\mu}{c} - g'(x) \\
+    /// b'(x) = -y'(x) \\
+    /// v'(x) = \frac{b(x) \cdot a'(x) - a(x) \cdot b'(x)}{b(x)^2}
+    /// $$
+    ///
+    /// And finally, the price after long derivative is:
+    ///
+    /// $$
+    /// p'(x) = v'(x) \cdot t_{s} \cdot v(x)^(t_{s} - 1)
+    /// $$
+    ///
+    fn price_after_long_derivative(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Result<FixedPoint> {
+        // g'(x)
+        let gov_fee_derivative = self.governance_lp_fee()
+            * self.curve_fee()
+            * (fixed!(1e18) - self.calculate_spot_price());
+
+        // a(x) = mu * (z_{e,0} + x/c - g(x))
+        let inner_numerator = self.mu()
+            * (self.ze() + base_amount / self.vault_share_price()
+                - self.open_long_governance_fee(base_amount));
+        // - self.zeta().into());
```

Comment:
Ok, I don't see the $\zeta$ term here, but it's still in the comments above. Unless there's something wrong with $self.ze()$ we shouldn't need it.